website/docs/concepts/event-bridge/index.md

```diff
@@ -28,6 +26,262 @@ To jump right into building, see the [`EventBus`](./event-bus.md) documentation.
+  (event) =>
+    event["detail-type"] === "Create" &&
+    event.detail.interests.includes("CATS") &&
+    event.detail.age >= 18 &&
+    event.detail.age < 30
+);
+```
+
+Rules can be further refined by calling `when` on a Functionless `Rule`.
+
+```ts
+// Cat people who are between 18 and 30 and do not also like dogs.
+catPeopleEvents.when((event) => !event.detail.interests.includes("DOGS"));
+```
+
+## Transform the event before sending to some services like `Lambda` Functions.
+
+We have two lambda functions to invoke, one for create or updates and another for deletes, lets make those.
+
+```ts
+const createOrUpdateFunction = new aws_lambda.Function(this, 'createOrUpdate', ...);
+const deleteFunction = new aws_lambda.Function(this, 'delete', ...);
+```
+
+and wrap them with Functionless's `Function` wrapper, including given them input types.
+
+```ts
+interface CreateOrUpdate {
+  id?: string;
+  name: string;
+  age: number;
+  operation: "Create" | "Update";
+  interests: string[];
+}
+
+interface Delete {
+  id: string;
+}
+
+const createOrUpdateOperation = functionless.Function<CreateOrUpdate, void>(
+  createOrUpdateFunction
+);
+const deleteOperation = functionless.Function<Delete, void>(deleteFunction);
+```
+
+The events from before do not match the formats from before, so lets transform them to the structures match.
+
+```ts
+const createOrUpdateEventsTransformed =
+  createOrUpdateEvents.map<CreateOrUpdate>((event) => ({
+    id: event.detail.id,
+    name: event.detail.name,
+    age: event.detail.age,
+    operation: event["detail-type"],
+    interests: event.detail.interests,
+  }));
+
+const deleteEventsTransformed = createOrUpdateEvents.map<Delete>((event) => ({
+  id: event.detail.id,
+}));
+```
+
+## Target other AWS services like Lambda and other Event Buses
+
+Now that we have created rules on our event buses using `when` and transformed those matched events using `map`, we need to send the events somewhere.
+
+We can `pipe` the transformed events to the lambda functions we defined earlier.
+
+```ts
+createOrUpdateEventsTransformed.pipe(createOrUpdateOperation);
+deleteEventsTransformed.pipe(deleteOperation);
+```
 
-There is a default Event Bus in every region of an AWS account. It contains events emitted by your AWS Resources, such as when a Step Function execution completes, or when a scheduled trigger fires.
+What about our young cat lovers? We want to forward those events to our sister team's event bus for processing.
+
+```ts
+const catPeopleBus = functionless.EventBus.fromBus(
+  aws_events.EventBus.fromEventBusArn(this, "catTeamBus", catTeamBusArn)
+);
+
+// Note: EventBridge does not support transforming events which target other event buses. These events are sent as is.
+catPeopleEvents.pipe(catPeopleBus);
+```
+
+## Put Events from other sources
+
+Event Bridge Put Events API is one of the methods for putting new events on an event bus. We support some first party integrations between services and event bus.
```

Comment:
`Functionless supports integrations between some AWS services and Event Bridge using the PutEvents API`